app.rb

```diff
@@ -0,0 +1,161 @@
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, name, age, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, name, age)
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here.'
+      return
+    end
+
+    puts 'Enter the number of the book you want: '
+    list_books
+    book_number = gets.chomp.to_i - 1
+
+    puts 'Enter your ID: '
+    list_people
+    person_id = gets.chomp.to_i
+    individual = @people.find { |person| person.id == person_id }
+
+    print 'Enter the date [yyyy-mm-dd]: '
+    date = gets.chomp.to_s
+    rental = Rental.new(date, @books[book_number], individual)
+    @rentals << rental
+
+    puts 'Book rented successfully.'
+  end
```

Comment:
- [ ] Good job so far 👏 However, as [suggested by the reviewer](https://github.com/axcell-0/school-library/pull/4#discussion_r1366199339), kindly make sure that the `create_rental` method is working and we are able to create a rental successfully 💪 Currently, I'm encountering the following error when trying to do so ⚠👇. 
- [ ] Since I am unable to create a rental, therefore I'm unable to test the `list_all_rentals` method. Please make sure that it is also working as expected as [the previous reviewer mentioned](https://github.com/axcell-0/school-library/pull/4#:~:text=please%20work%20on%20this%20and%20also%20ensure%20that%20the%20list%20rentals%20method%20also%20works%20as%20expected) ⛏

![image](https://github.com/axcell-0/school-library/assets/106119105/7e930c09-6359-4072-a309-06268f4801a3)

_You could try something like below 👇 Kindly make sure to run your app before asking for another review_

 ```ruby
puts 'Enter the number of the book you want: '
list_books
book_number = gets.chomp.to_i - 1

if book_number < 0 || book_number >= @books.size
  puts 'Invalid book number. Please try again.'
  return
end

puts 'Enter the number of the person you want to rent to: '
list_people
person_number = gets.chomp.to_i - 1

if person_number < 0 || person_number >= @people.size
  puts 'Invalid person number. Please try again.'
  return
end

print 'Enter the date [yyyy-mm-dd]: '
date = gets.chomp.to_s
rental = Rental.new(date, @books[book_number], @people[person_number])
@rentals << rental
```


- [OPTIONAL] I noticed that the `age` and `name` values are interchanged in listing people. Could you please fix this one as well? ✨💪

![image](https://github.com/axcell-0/school-library/assets/106119105/e87c0b1b-f1cb-4de0-8a62-9141a00c515c)

Reply:
@KanzaTahreem Thank you very much for the review I do appreciate for the kind counselling, but please on my machine the create rental method is working please can you help me to understand cause this is the following rental I created on my machine
![image](https://github.com/axcell-0/school-library/assets/121089024/eb3e9f6c-76b0-4b90-8ca0-ecbbd49fa3b1)

Reply:
Hello @axcell-0 great job ✨But I noticed that you are creating a rental by passing a person's ID, however, we need to pass the no.(1, 2, 3 etc), not the ID, while creating a rental. Please have a look at the [image provided previously](https://github.com/axcell-0/school-library/pull/4#discussion_r1366199339)